### ISSUE TITLE
Pull common functionality out of treeburner into umbra

### ISSUE THREAD

**Issue by odiminox** [libtcod/umbra#1]
treeburner contains a large collection of utilities and objects that can support the development of any roguelike game. The capability of umbra could be expanded into a general rougelike engine/toolkit by refactoring these tools out of treeburner into umbra

**Comment by HexDecimal** [comment]
What I was doing was working on Umbra and it's sub-projects simultaneously, but this sounds like a better idea.  The 3 projects using Umbra all share similar codebases, so it might work better to move the duplicate code to Umbra itself.

**Comment by odiminox** [comment]
@HexDecimal I presume we want to refactor umbra code into namespaces as I have done on treeburner

**Comment by odiminox** [comment]
That way treeburner code pulled into Umbra can be easily ported

**Comment by HexDecimal** [comment]
Yeah.  Especially since this is supposed to be included into other projects.  All these headers used to have an `umbra_` prefix before, but I didn't add a namespace.

I'd probably go with `umbra::` for everything.  Maybe also `umbra::widget::`.  Not sure about `imod`, that needs to be unabbreviated.
